### PR TITLE
doc: introduction: distinguishing NCS features

### PR DESCRIPTION
**Pull request by greg-fer**
Edited the introduction page to include distinguishing features similar to the section in Zephyr docs.
NCSDK-22322.

---- 

- [x] Approval from @wbober

### CODE REVIEW

File: doc/nrf/introduction.rst
Comment (wbober) [inline doc/nrf/introduction.rst:32]:
```suggestion
   The |NCS| provides integrations with third-party and Nordic products within the SDK, such as AWS, nRFCloud, memfault, and more.
```

File: doc/nrf/introduction.rst
Comment (wbober) [inline doc/nrf/introduction.rst:29]: "In addition to connectivity technologies provided by Zephyr [list] it supports [list]".

File: doc/nrf/introduction.rst
Comment (wbober) [inline doc/nrf/introduction.rst:26]: Provided pre-certified and optimized libraries.

File: doc/nrf/introduction.rst
Comment (wbober) [inline doc/nrf/introduction.rst:23]: I'd remove this.

File: doc/nrf/introduction.rst
Comment (wbober) [inline doc/nrf/introduction.rst:20]: Put as first.

File: doc/nrf/introduction.rst
Comment (wbober) [inline doc/nrf/introduction.rst:17]: In addition to middleware provided by Zephyr, adds extra libraries such as [list only these that are not available upstream].

Comment (greg-fer) [reply doc/nrf/introduction.rst]: Added some, but would appreciate a more thorough look at this.

**Review by DeidreCasey — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by greg-fer** [comment]
@wbober , I will wait for merge request for your sign-off on this one.

**Review by wbober — APPROVED** [review]

**Review by b-gent — APPROVED** [review]

**Pull request merged**
